automata/fa/animation.py

```diff
@@ -0,0 +1,376 @@
+        highlighted.
+
+        Parameters
+        ----------
+        old_states : Iterable[FAStateT]
+            The states to turn to default color.
+        new_states : Iterable[FAStateT]
+            The states to highlight.
+
+        Returns
+        -------
+        Iterable[ApplyMethod]
+            The animations for the `Scene` object to `play`.
+        """
+        yield from (
+            Animate.to_default_color(self[old_state]) for old_state in old_states
+        )
+        yield from self.highlight_states(new_states)
+
+    def change_transitions(
+        self,
+        old_transitions: Iterable[tuple[FAStateT, FAStateT]],
+        new_transitions: Iterable[tuple[FAStateT, FAStateT]],
+    ) -> Iterable[manim.ApplyMethod]:
+        """
+        Turn `old_transitions` to default color and highlight the `new_transitions`.
+
+        Transitions occured in both `old_transitions` and `new_transitions` will
+        still be highlighted.
+
+        Parameters
+        ----------
+        old_transitions : Iterable[FAStateT]
+            The transitions to turn to default color.
+        new_transitions : Iterable[FAStateT]
+            The transitions to highlight.
+
+        Returns
+        -------
+        Iterable[ApplyMethod]
+            The animations for the `Scene` object to `play`.
+        """
+        yield from (
+            Animate.to_default_color(self[old_transition])
+            for old_transition in old_transitions
+        )
+        yield from (
+            Animate.highlight(self[new_transition])
+            for new_transition in new_transitions
+            if new_transition[-1] is not None
+        )
+
+    def clean(
+        self, transitions: Iterable[tuple[FAStateT, FAStateT]]
+    ) -> Iterable[manim.ApplyMethod]:
+        """
+        Cancel all the highlighted elements.
+
+        Parameters
+        ----------
+        transitions : Iterable[tuple[FAStateT, FAStateT]]
+            The highlighed transitions to cancel the highlight.
+
+        Returns
+        -------
+        The animations for `Scene` object to `play`.
+        """
+        yield from (
+            Animate.to_default_color(self[transition])
+            for transition in transitions
+            if transition[1] is not None
+        )
+
+
+@dataclass(eq=False)
+class _DFAAnimation(manim.Scene):
+    """
+    The `_DFAAnimation` class is the class to generate the animation of a DFA
+    identifying an input string.
+
+    To generate the animation, use `_DFAAnimation.render`, which will call the
+    `setup` method and the `construct` method.
+
+    Parameters
+    ----------
+    dfa : DFA
+        The DFA object.
+    input_str : str
+        The string to identify.
+    dfa_graph : _FAGraph
+        The graph of the DFA.
+    input_symbols : _ManimInput
+        The element in animation to show the current symbol of the `input_str`.
+    """
+
+    dfa: "DFA"
```

Comment:
I think the string is not needed if you have `from __future__ import annotations`